content/guides/development.md

```diff
@@ -5,182 +5,247 @@ contributors:
 
-T> You may find the `--single` option useful for serving single page apps.
 
+## Choosing a Development Tool
 
-### Watch Mode with Chrome DevTools Workspaces
+W> Some text editors have a "safe write" function that might interfere with some of the following tools. Read [`Adjusting Your text Editor`](#adjusting-your-text-editor) for a solution to these issue.
 
-If you set up Chrome to [persist changes when saving from the _Sources_ panel](https://medium.com/@rafaelideleon/webpack-your-chrome-devtools-workspaces-cb9cca8d50da)
-so you don't have to refresh the page, you will have to setup webpack to use
+It quickly becomes a hassle to manually run `npm run build` everytime you want to compile your code.
 
-```javascript
-devtool: "inline-source-map"
-```
+There are a couple of different options available in webpack that help you automatically compile your code whenever it changes:
 
-to continue editing and saving your changes from Chrome or source files.
+ 1. webpack's Watch Mode
+ 2. webpack-dev-server
+ 3. webpack-dev-middlware
 
-There are some _gotchas_ about using workspaces with watch:
+In most cases, you probably would want to use `webpack-dev-server`, but let's explore all of the above options.
 
-* Large chunks (such as a common chunk that is over 1MB) that are rebuilt could cause the page to blank, which will force you to refresh the browser.
-* Smaller chunks will be faster to build than larger chunks since `inline-source-map` is slower due to having to base64 encode the original source code.
 
+### Using Webpack Watch Mode
 
-### webpack-dev-server
+Webpack's Watch Mode allows you to watch a set of files for changing. Webpack will then automatically recompile your code for you, so you don't have to do this manually.
 
-webpack-dev-server provides you with a server and live reloading. This is easy to setup.
+Let's add an npm script that will start webpack's Watch Mode:
 
-To prepare, make sure you have a `index.html` file that points to your bundle. Assuming that `output.filename` is `bundle.js`:
+__package.json__
 
-```html
-<script src="/bundle.js"></script>
+``` diff
+  {
+    "name": "asset-management",
+    "version": "1.0.0",
+    "description": "",
+    "main": "webpack.config.js",
+    "scripts": {
+      "test": "echo \"Error: no test specified\" && exit 1",
++     "watch": "webpack --progress --watch",
+      "build": "webpack"
+    },
+    "keywords": [],
+    "author": "",
+    "license": "ISC",
+    "devDependencies": {
+      "css-loader": "^0.28.4",
+      "csv-loader": "^2.1.1",
+      "file-loader": "^0.11.2",
+      "html-webpack-plugin": "^2.29.0",
+      "style-loader": "^0.18.2",
+      "webpack": "^3.0.0",
+      "xml-loader": "^1.2.1"
+    }
+  }
 ```
 
-Start with installing `webpack-dev-server` from npm:
+Now you can run `npm run watch` from the command line and you will see that webpack compiles your code, but it doesn't exit to the command line (this is because the script is still watching your files).
```

Comment:
Probably better as:

> You can now run `npm run watch` from the command line to see that webpack compiles your code, but doesn't exit to the command line. This is because the script is still watching your files.